cooltools/expected.py

```diff
@@ -417,101 +519,124 @@ def trans_expected(clr, chromosomes, chunksize=1000000, use_dask=False):
-    ntot = (
-        n_total_trans_elements(clr, chromosomes)
-        .groupby(("chrom1", "chrom2"))["n_total"]
-        .sum()
-    )
-    nbad = (
-        n_bad_trans_elements(clr, chromosomes)
-        .groupby(("chrom1", "chrom2"))["n_bad"]
-        .sum()
-    )
-    trans_area = ntot - nbad
-    trans_area.name = "n_valid"
-    # processing with use_dask=True is different:
-    if use_dask:
-        # trans_sum = pixels.groupby(['chrom1', 'chrom2'])['balanced'].sum().compute()
-        pass
-    else:
-        trans_sum = pixels.groupby(["chrom1", "chrom2"])["balanced"].sum()
-    # for consistency with the cis_expected function:
-    trans_sum.name = trans_sum.name + ".sum"
-
-    # returning a DataFrame with MultiIndex, that stores
-    # pairs of 'balanced.sum' and 'n_valid' values for each
-    # pair of chromosomes.
-    dtable = pd.merge(
-        trans_sum.to_frame(), trans_area.to_frame(), left_index=True, right_index=True
-    )
+
+    # turn chromosomes into supports:
+    chrom_supports = [ (chrom, 0, None) for chrom in chromosomes ]
+    # use balaned transformation only:
+    balanced_transform = {
+            "balanced": \
+                lambda pixels: pixels["count"] * pixels["weight1"] * pixels["weight2"]
+            }
+
+    # trans_expected is simply a wrapper around new blocksum_pairwise
+    # but it preserved the interface of the original trans_expected
+    trans_records = blocksum_pairwise(clr,
+                            supports=chrom_supports,
+                            transforms=balanced_transform,
+                            chunksize=chunksize)
+
+    # trans_records are inter-chromosomal only,
+    # changing trans_records keys to reflect that:
+    # region[0] for a region = (chrom, start, stop)
+    trans_records = {
+        ( region1[0], region2[0] ): val for ( region1, region2 ), val in trans_records.items()
+        }
+
+    # turn trans_records into a DataFrame with
+    # MultiIndex, that stores values of 'balanced.sum'
+    # and 'n_valid' values for each pair of chromosomes:
+    trans_df = pd.DataFrame.from_dict( trans_records, orient="index" )
+    trans_df.index.rename( ["chrom1","chrom2"], inplace=True )
+    # an alternative way to get from records to DataFrame, as in CLI expected:
+    # result = pd.DataFrame(
+    #     [
+    #         {"chrom1": s1[0], "chrom2": s2[0], **rec}
+    #         for (s1, s2), rec in trans_records.items()
+    #     ],
+    #     columns=["chrom1", "chrom2", "n_valid", "count.sum", "balanced.sum"],
+    # )
 
     # the actual expected is balanced.sum/n_valid:
-    dtable["balanced.avg"] = dtable["balanced.sum"] / dtable["n_valid"]
-    return dtable
+    trans_df["balanced.avg"] = trans_df["balanced.sum"] / trans_df["n_valid"]
+    return trans_df
 
 
 ###################
 
 
-def make_diag_tables(clr, supports):
+def make_diag_tables(clr, supports, weight_name="weight", bad_bins=None):
```

Comment:
`weight_name="weight", bad_bins=None` added here and below
`bad_bins` is just a placeholder for now